From 28f44c98ccdd97756e4d2d6a01b7cfa3a34cfa22 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 1 Nov 2017 22:24:54 +0800
Subject: [PATCH 7/8] video: sunxi: de2: add SimpleFB node creation for R40

R40 features different display pipelines than other SoCs with DE2.

Add SimpleFB node creation code for it.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/video/sunxi/sunxi_de2.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/sunxi/sunxi_de2.c b/drivers/video/sunxi/sunxi_de2.c
index dc6ceea75e..2b1b661d51 100644
--- a/drivers/video/sunxi/sunxi_de2.c
+++ b/drivers/video/sunxi/sunxi_de2.c
@@ -323,17 +323,21 @@ int sunxi_simplefb_setup(void *blob)
 	struct udevice *de2, *hdmi, *lcd;
 	struct video_priv *de2_priv;
 	struct video_uc_platdata *de2_plat;
+#ifndef CONFIG_MACH_SUN8I_R40
 	int mux;
+#endif
 	int offset, ret;
 	u64 start, size;
 	const char *pipeline = NULL;
 
 	debug("Setting up simplefb\n");
 
+#ifndef CONFIG_MACH_SUN8I_R40
 	if (IS_ENABLED(CONFIG_MACH_SUNXI_H3_H5))
 		mux = 0;
 	else
 		mux = 1;
+#endif
 
 	/* Skip simplefb setting if DE2 / HDMI is not present */
 	ret = uclass_find_device_by_name(UCLASS_VIDEO,
@@ -348,10 +352,14 @@ int sunxi_simplefb_setup(void *blob)
 	if (ret) {
 		debug("HDMI not present\n");
 	} else if (device_active(hdmi)) {
+#ifndef CONFIG_MACH_SUN8I_R40
 		if (mux == 0)
 			pipeline = "mixer0-lcd0-hdmi";
 		else
 			pipeline = "mixer1-lcd1-hdmi";
+#else
+		pipeline = "mixer0-tcon_tv0-hdmi";
+#endif
 	} else {
 		debug("HDMI present but not probed\n");
 	}
-- 
2.13.6

